src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -4494,7 +4494,9 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
         val opeqStart = if (Statistics.canEnable) Statistics.startTimer(failedOpEqNanos) else null
 
         def onError(reportError: => Tree): Tree = fun match {
-          case Select(qual, name) if !mode.inPatternMode && nme.isOpAssignmentName(newTermName(name.decode)) =>
+          case Select(qual, name)
+            if !mode.inPatternMode && nme.isOpAssignmentName(newTermName(name.decode)) && !qual.exists(_.isErroneous) =>
```

Comment:
I tried that but it is not. In the test case, the `Apply` node itself has no error. The tree looks like:

```
Apply(Ident(TermName("c")), List(Ident(<TermName("undefinedSymbol"): error>)))
```